src/Features/Core/Portable/MakeMethodAsynchronous/AbstractMakeMethodAsynchronousCodeFixProvider.cs

```diff
@@ -47,27 +47,21 @@ internal abstract class AbstractMakeMethodAsynchronousCodeFixProvider : CodeFixP
             }
 
             var symbol = semanticModel.GetDeclaredSymbol(node, cancellationToken) as IMethodSymbol;
+            bool isEntryPoint = symbol != null && symbol.IsStatic && string.Equals(symbol.Name, "Main", StringComparison.Ordinal);
```

Comment:
Considering the large conversation below about performance, should this at least do a _quick_ check to see if CompilationOptions.MainTypeName is set, and if so, bring that into consideration here?

Reply:
I think you are proposing:
1. if MainTypeName is set, then do expensive check
2. otherwise, use heuristic (cheap check on method to see if it looks like `Main`)

I'm ok to rely purely on heuristic (only doing (2) above), since using an entry point other than `Main` seems uncommon.